examples/upbound-aws-provider/composite-resources/database-examples/aurora/README.md

```diff
@@ -0,0 +1,240 @@
+
+```
+    (It should take about 15-20 min to provision the Aurora RDS cluster.)
+    
+    Below is the default behaviour of the resource which will be provisioned through the claim, just to mention all this default behaviour can be overridden through patching .
+
+    1. It will create a 3 node cluster with 1 writer and 2 reader endpoint spread across 3 Azs by default.
+    2. It will also create a proxy to connect to the aurora cluster.
+    3. The management of database credential is done through the secret manager.
+    4. The security group has 2 rules , one to allow the app to connect to the proxy and one within the security group
+        where the proxy can connect to the database (Make sure you have provided the correct the CIDR at the claim which is the allow CIDR on the security group from app.)
+    5. The Aurora DB has been configured with logging and monitoring.
+
+ ### Steps to check the connectivity from a pod 
+
+Create an  policy to allow the pod to connect to the Aurora Database
+  
+```shell
+    aws iam create-role --role-name aurora-monitoring \
+                        --assume-role-policy-document '{    
+        "Version": "2012-10-17",
+        "Statement": [
+        {
+            "Sid": "",
+            "Effect": "Allow",
+                Action": [
+                    "rds-db:connect"
+                ],
+                "Resource": [
+                    "arn:aws:rds-db:${REGION_NAME}>:${ACCOUNT_NUMBER}:dbuser:*/*"
+                ]
+            }
+        ]
+    }'
+```
+    
+    Create a service account passing the above policy arn
+
+ ```shell
+        eksctl create iamserviceaccount \
+        --name rds-access \
+        --namespace team-a \
+        --cluster ${CLUSTER_NAME} \
+        --attach-policy-arn ${POLICY_ARN} \
+        --approve \
+        -override-existing-serviceaccounts
+``` 
+    Generate a DB token which can be used for password when asked
+    
+```shell
+    
+    PROXY_TERGET-ENDPOINT=k get secrets aurora-cluster-secrets  -n team-a -o json | jq -r .data.proxyEndpoint | base64 --decode
+    CLUSTER_USER_NAME=k get secrets aurora-cluster-secrets  -n team-a -o json | jq -r .data.clusterUsername | base64 --decode
+    
+    aws rds generate-db-auth-token \
+    --hostname ${PROXY_TERGET-ENDPOINT}  \
+    --port 5432 \
+    --region us-east-1 \
+    --username ${CLUSTER_USER_NAME}
+    
+```
+   create the pod with pgsql client:
+
+```shell
+
+    apiVersion: v1
+    kind: Pod
+    metadata:
+    name: postgres-client
+    namespace: team-a
+    spec:
+    serviceAccountName: rds-access
+    containers:
+    - name: postgreclient
+        image: postgres:latest
+        command: ["sleep"]
+        args: ["3600"]  # Sleep for 1 hour (3600 seconds)
+        envFrom:
+        - secretRef:
+            name: aurora-cluster-secrets
+    volumes:
+        - name: secret-volume
+        secret:
+            secretName: aurora-cluster-secrets # This secret name can be changed in the claim
```

Comment:
I would rather put this in a file so users can just `k apply -f file`

Reply:
moved to a different file..